metaflow/plugins/kfp/spot_decorator.py

```diff
@@ -0,0 +1,24 @@
+from metaflow.decorators import StepDecorator
+
+
+class SpotDecorator(StepDecorator):
+    """
+    For KFP orchestrator plugin only.
+
+    Step decorator to specify that the pod be on a Spot node.
+
+    To use, follow the example below.
+    ```
+    @spot()
```

Comment:
Playing devils advocate here, do we ever think there's a use case to use spot in one environment but on-demand in another? If so what would the syntax be to support that?

With this approach the declaration makes it uniform that all environments must use spot, as opposed to an alternative syntax (ie. `@resources(..., spot=True)` or `@resources(..., capacity_type="spot")`) where you could use a `boolean` to swap this out.

Reply:
I assume with spot decorator we can also do `--with spot` based on CICD template vars.

We correctly don't have a way to change resource specification per env, but we should build one in the near future.

Reply:
Putting on the Metaflow design hat, `@spot()`, means use Spot if available in the environment and if there are spot instances available, else use on-demand.

Now, where should the request for SPOT be (putting on the Metaflow design hat)? 
1. A `@spot` decorator
2. In the compute decorator, such as:
    a. https://github.com/zillow/metaflow/blob/9ca35b6a29ce0544bde38f66cf8017925457cc9b/metaflow/plugins/aws/batch/batch_decorator.py#L28
   b. https://github.com/zillow/metaflow/blob/9ca35b6a29ce0544bde38f66cf8017925457cc9b/metaflow/plugins/aws/eks/kubernetes_decorator.py#L24
   c. https://github.com/zillow/metaflow/blob/9ca35b6a29ce0544bde38f66cf8017925457cc9b/metaflow/plugins/kfp/kfp_decorator.py#L27 

3. In the `@resources()` decorator.  

I think that putting it in `@spot` is a good first step towards declaring intent by the author that this step is interruptible, and that the author has codified checkpoints and resume-able behavior.

This is a first attempt at the SDK, and is experimental.  Longer-term (end of Q4 or Q1 next year) I believe the decorator may morph into a `@interruptable` as that better declares intent.  It would then be the Platform policy on how to save costs, A) use SPOT if available by orchestrator and environment and B) defrag/consolidation if available [1]

[1] Although how and when defrag interruptions would be acceptable is to be determined, most likely at checkpoint time.  However, we may allow N interruptions outside of checkpoint time to save costs (hopefully a Platform parameter).

Reply:
Personally I like the idea of `@interruptable` far better, any reason we cannot just implement that now?

Reply:
done!